frontend/src/components/Aggregate.tsx

```diff
@@ -28,11 +67,11 @@ const Aggregate = () => {
           </div>
           <input
             type="number"
-            value={numSemesters}
+            value={numSemsField === undefined ? "" : numSemsField}
             className="border-gray-200 min-w-0 flex-auto rounded border px-2 py-1 text-sm bg-transparent"
-            onChange={(e) => {
-              setNumSemesters(parseInt(e.target.value));
-            }}
+            onChange={handleNumSemsFieldChange}
+            onBlur={handleNumSemsFieldBlur}
+            onKeyPress={handleNumSemsFieldKeyDown}
           />
```

Comment:
Also, could you refactor the numSems input out as another component in the same file? The logic is getting a bit complicated.